From 65fb8284564a5d6db8c4884cad903119d85dce3f Mon Sep 17 00:00:00 2001
From: Himal Prasad Ghimiray <himal.prasad.ghimiray@intel.com>
Date: Mon, 3 Aug 2026 10:37:19 +0530
Subject: drm/xe/dma-buf: propagate pin error

Return the error from xe_bo_pin_external() instead of asserting
success, so callers observe failures such as -ENOSPC from the pin limit.

Reviewed-by: Tejas Upadhyay tejas.upadhyay@intel.com
Signed-off-by: Himal Prasad Ghimiray <himal.prasad.ghimiray@intel.com>
---
 src/drivers/gpu/drm/xe/xe_dma_buf.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/gpu/drm/xe/xe_dma_buf.c b/drivers/gpu/drm/xe/xe_dma_buf.c
index de6999f89..15a2290b4 100644
--- a/drivers/gpu/drm/xe/xe_dma_buf.c
+++ b/drivers/gpu/drm/xe/xe_dma_buf.c
@@ -80,10 +80,15 @@ static int xe_dma_buf_pin(struct dma_buf_attachment *attach)
 		}
 	}
 
+#ifdef BPM_DRMM_CGROUP_REGISTER_REGION_NOT_PRESENT
+	return xe_bo_pin_external(bo, !allow_vram, exec);
+#else
 	ret = xe_bo_pin_external(bo, !allow_vram, exec);
 	xe_assert(xe, !ret);
 
 	return 0;
+#endif
+
 }
 
 static void xe_dma_buf_unpin(struct dma_buf_attachment *attach)
-- 
2.43.0

